/*
 * Copyright (c) 2014 Juniper Networks, Inc. All rights reserved.
 */

import "analytics/protobuf_schema.proto";

message TestMessageInner {
    required string tm_inner_name = 1 [(telemetry_options).is_key = true];
    optional int32 tm_inner_status = 2;
    optional int32 tm_inner_counter = 3;
    optional TestMessage.TestMessageEnum tm_inner_enum = 4 [(telemetry_options).is_key = true];
}

message TestMessage {
    required string tm_name = 1 [(telemetry_options).is_key = true];
    optional string tm_status = 2 [(telemetry_options).is_key = true];
    optional int32 tm_counter = 3 [(telemetry_options).is_key = true];
    repeated TestMessageInner tm_inner = 4;
    enum TestMessageEnum {
        GOOD = 1;
        BAD = 2;
    }
    optional TestMessageEnum tm_enum = 5 [(telemetry_options).is_key = true];
}

message TestMessageSize {
    required bytes tms_data = 1;
}

message TestMessageBase {
    required string tmb_name = 1 [(telemetry_options).is_key = true];
    optional string tmb_status = 2 [(telemetry_options).is_key = true];
    optional int32 tmb_counter = 3 [(telemetry_options).is_key = true];
    enum TestMessageEnum {
        GOOD = 1;
        BAD = 2;
    }
    optional TestMessageEnum tmb_enum = 4 [(telemetry_options).is_key = true];
    extensions 10 to max;
}

message TestMessageContrailSensors {
    extensions 1 to max;
}

message TestMessageBaseSensors {
    optional TestMessageContrailSensors tmbs_sensor = 2636;
    extensions 1 to 2635;
    extensions 2637 to max;
}

message TestMessageBaseStream {
    required string tmbs_name = 1 [(telemetry_options).is_key = true];
    optional TestMessageBaseSensors tmbs_sensor = 101;
}
